ci(wasm): stale nightly toolchain

## Symptom

Every push to `main` shows `test_halo2_wasm_web (halo2)` failing in Build and Test, while its dependency `cli_build_web (halo2)` reports success. Example: https://github.com/zkmopro/mopro/actions/runs/26563792364/job/78254980996

The failing step is `Install dependencies`:

```
##[error]An error occurred trying to start process '/usr/bin/bash' with working directory
'/home/runner/work/_temp/mopro-example-halo2/web'. No such file or directory
```

This regression isn't new — it's been red on `main` since at least 2026-03-16. It was just less visible because the actual cause is buried in the upstream `cli_build_web` log, not in the job that fails.

## Root cause (two coupled bugs)

### 1. Pinned wasm nightly toolchain is too old

`mopro-ffi/rust-toolchain.toml:2`, `mopro-ffi/src/app_config/web.rs:47`, and `.github/workflows/build-and-test.yml:525,557` all pin `nightly-2024-07-18` (cargo 1.81.0-nightly, July 16 2024). That toolchain predates `edition2024`.

A transitive dep — `wit-bindgen 0.51.0` / `wit-bindgen-rust-macro 0.51.0` — uses `edition = "2024"`, so cargo refuses to parse its manifest:

```
error: failed to parse manifest at .../wit-bindgen-rust-macro-0.51.0/Cargo.toml
Caused by:
  feature `edition2024` is required
  The package requires the Cargo feature called `edition2024`, but that feature is not stabilized
  in this version of Cargo (1.81.0-nightly (a2b58c3da 2024-07-16)).
```

`wasm-pack build` exits non-zero, `mopro-ffi/src/app_config/web.rs` prints `mopro-ffi wasm package build failed.` and calls `std::process::exit(1)`.

### 2. The `Build Web` step swallows the non-zero exit code

`.github/workflows/build-and-test.yml:530` wraps `mopro build` in a util-linux `script` invocation to fake a TTY for the interactive prompt:

```yaml
printf 'y\n' | script -q -c "$GITHUB_WORKSPACE/bin/mopro build --mode release --platforms web" /dev/null
```

Util-linux `script` returns **its own** exit code (0) by default, **not** the child's, unless `-e` / `--return` is passed. So mopro exits 1, `script` exits 0, and the `Build Web` step reports success.

Cascade from there:
- `actions/cache@v4` skips the save with `Path Validation Error: Path(s) specified in the action for caching do(es) not exist, hence no cache is being saved.` (no `MoproWasmBindings/` was produced).
- `test_halo2_wasm_web` runs, cache-misses, finds no `web/` directory, and `yarn install` blows up with "No such file or directory" — the surface failure the user sees.

This is a silent-failure foot-gun: any future regression in `mopro build --platforms web` will look like a downstream cache-miss instead of a build error.

## Proposed fix

One PR with three small changes:

1. Bump `nightly-2024-07-18` → `nightly-2025-02-20` (around Rust 1.85 stabilization, when `edition2024` became stable). Four files:
   - `mopro-ffi/rust-toolchain.toml:2`
   - `mopro-ffi/src/app_config/web.rs:47`
   - `.github/workflows/build-and-test.yml:525`
   - `.github/workflows/build-and-test.yml:557`
2. Add `-e` to the Linux `script` invocation in `build-and-test.yml:530` so a failing wasm build fails the step. Leave the macOS `script -q /dev/null cmd args` form alone — BSD `script` semantics differ and the macOS web path isn't currently broken.
3. Defense in depth: after the Build Web step, assert `test -d MoproWasmBindings` and `exit 1` with a `::error::` annotation if it's missing. Catches any future silent-exit class of bug at the step that introduced it.

A draft of the patch is sitting locally on branch `fix/wasm-ci-nightly-and-silent-exit` — happy to push it as a PR once we agree on the toolchain version.

## Open questions

- Is `nightly-2025-02-20` the right pin, or do we want something more recent / something specific that's known to work with the wasm-pack + `-C target-feature=+atomics,+bulk-memory` flags? Any in-tree feature that requires a specific nightly?
- Worth fixing the underlying need for `script` (`# TODO: allow mopro CLI to run fully non-interactive`) as a follow-up? That would let us drop the pseudo-TTY entirely instead of bandaging it with `-e`.

🤖 Drafted with [Claude Code](https://claude.com/claude-code)

--- a/.github/workflows/build-and-test.yml
+++ b/.github/workflows/build-and-test.yml
@@ -522,12 +522,16 @@ jobs:
                   CI: "true"
               run: |
                   set -euo pipefail
-                  rustup component add rust-src --toolchain nightly-2024-07-18
+                  rustup component add rust-src --toolchain nightly-2025-02-20
                   # TODO: allow mopro CLI to run fully non-interactive so we can drop the pseudo-tty workaround below
                   if [[ "$RUNNER_OS" == "macOS" ]]; then
                     printf 'y\n' | script -q /dev/null "$GITHUB_WORKSPACE/bin/mopro" build --mode release --platforms web
                   else
-                    printf 'y\n' | script -q -c "$GITHUB_WORKSPACE/bin/mopro build --mode release --platforms web" /dev/null
+                    printf 'y\n' | script -q -e -c "$GITHUB_WORKSPACE/bin/mopro build --mode release --platforms web" /dev/null
+                  fi
+                  if [ ! -d MoproWasmBindings ]; then
+                    echo "::error::wasm build did not produce MoproWasmBindings/"
+                    exit 1
                   fi
             - name: Create Web framework (release, halo2)
               working-directory: ${{ runner.temp }}/mopro-example-${{ matrix.adapter }}
@@ -554,7 +558,7 @@ jobs:
             - name: Setup Rust
               uses: ./.github/actions/setup-rust
               with:
-                  toolchain: nightly-2024-07-18
+                  toolchain: nightly-2025-02-20
                   cache-key: wasm-env
             - name: Cache wasm-pack
               id: cache-wasm-pack

--- a/mopro-ffi/rust-toolchain.toml
+++ b/mopro-ffi/rust-toolchain.toml
@@ -1,4 +1,4 @@
 [toolchain]
-channel = "nightly-2024-07-18"
+channel = "nightly-2025-02-20"
 components = ["rust-src"]
 targets = ["wasm32-unknown-unknown"]

--- a/mopro-ffi/src/app_config/web.rs
+++ b/mopro-ffi/src/app_config/web.rs
@@ -44,7 +44,7 @@ impl PlatformBuilder for WebPlatform {
         let mut cmd = Command::new("rustup");
         cmd.args([
             "run",
-            "nightly-2024-07-18",
+            "nightly-2025-02-20",
             "wasm-pack",
             "build",
             "--target",

